code/modules/vehicles/vehicle.dm

```diff
@@ -144,3 +147,28 @@
 
 /obj/vehicle/space/Process_Spacemove(direction)
 	return 1
+
+/obj/vehicle/run_obj_armor/run_obj_armor(damage_amount, damage_type, damage_flag = 0, attack_dir)
+	if(damage_flag == "melee" && damage_amount < 20)
+		return 0
+	. = ..()
+
+/obj/vehicle/deconstruct(disassembled = TRUE)
+	new /obj/item/stack/sheet/metal (loc, 5)
+	qdel(src)
+
+/obj/vehicle/examine(mob/user)
+	..()
+	if(!(resistance_flags & INDESTRUCTIBLE))
+		if(resistance_flags & ON_FIRE)
+			user << "<span class='warning'>It's on fire!</span>"
+		var/healthpercent = (obj_integrity/max_integrity) * 100
+		switch(healthpercent)
+			if(100 to INFINITY)
+				user <<  "It seems pristine and undamaged."
+			if(50 to 100)
+				user <<  "It looks slightly damaged."
+			if(25 to 50)
+				user <<  "It appears heavily damaged."
+			if(0 to 25)
+				user <<  "<span class='warning'>It's falling apart!</span>"
```

Comment:
I had to put some nearly copypasta for the examine() of vehicle/machinery/structure, because I can't have it at /obj/ level. It would appear on decals, smoke effect, and items which we don't want.